engine/client/gl_rmain.c

```diff
@@ -1326,90 +1328,245 @@ void R_RenderFrame( const ref_params_t *fd, qboolean drawWorld )
+	}
+	mean = mean / n;
+	for (i = 0; i < n; ++i)
+		sum_deviation += (data[i] - mean)*(data[i] - mean);
+	return sqrt(sum_deviation / n);
+}
+
 /*
 ===============
 R_Strobe
 
-TODO: Consider vsync timings and do not render the supposed black frame at all.
+TODO:
+	* (?) Move "Strobe" to separate files.
+	* Make swapping transition seamless by rendering non-opaque frames.
+	* Implement high precision timer to keep the internal phase on track with monitor. (In case of freezes etc.)
+	* (?) Call R_Strobe each frame no matter strobe setting and simulate strobing to keep the phases on track (When on menu, ...)
 ===============
 */
 void R_Strobe( void )
 {
-	static int sCounter = 0;
-	int getInterval = r_strobe->integer; // Check through modified tag first?
+	static int strobeInterval = 0;
+	static int swapInterval = 0;
+	static double recentTime = 0.0, recentTime2 = 0.0;
+	double _delta = 0.0, _delta2 = 0.0;
+	static qboolean cdTriggered = false;
+	static double delta[STROBE_DEVIATION_SIZE] = { 0.0 };
+	static unsigned int fCounterSnapshot = 0;
+	double currentTime = Sys_DoubleTime();
+	
+	_delta2 = currentTime - recentTime2;
+	recentTime2 = currentTime;
 
-	if( CL_IsInMenu() )
+	if (CL_IsInMenu())
 	{
 		R_Set2DMode(false);
 		return;
 	}
 
-	if( !getInterval || ( !gl_swapInterval->integer && getInterval ) )
+	if( StrobeInfo.cdTimer >= 0.0)
+		StrobeInfo.cdTimer += _delta2;
+	if (StrobeInfo.fCounter - fCounterSnapshot == 1)
 	{
-		if( getInterval ) //If v-sync is off, turn off strobing
+		delta[StrobeInfo.fCounter % ARRAYSIZE(delta)] = _delta2;
+		StrobeInfo.deviation = standard_deviation(delta, ARRAYSIZE(delta)) * 1000;
+	}
+	fCounterSnapshot = StrobeInfo.fCounter;
+	
+	
+	if (r_strobe_cooldown->integer > 0)
+	{
+		if (StrobeInfo.cdTimer > (double)abs(r_strobe_cooldown->integer) && cdTriggered == true)
+		{
+			cdTriggered = false;
+			StrobeInfo.cdTimer = -1.0;
+		}
+
+		if (StrobeInfo.fCounter > ARRAYSIZE(delta))
+		{
+			if (StrobeInfo.deviation > STROBE_DEVIATION_LIMIT)
+			{
+				cdTriggered = true;
+				StrobeInfo.cdTimer = 0.0;
+			}
+		}
+	}
+	else
+	{
+		cdTriggered = false;
+	}
+
+	//Msg("Snapshot: %d - Deviation: %f - timer %f - delta %f\n\n", snapshot, standard_deviation(data, 10) * 1000, StrobeInfo.cdTimer, _delta2);
+
+	if ( ((strobeInterval != r_strobe->integer) && (strobeInterval != 0) ) || 
+		/*((swapInterval != r_strobe_swapinterval->integer) && (swapInterval != 0)) || */
+		StrobeInfo.fCounter > UINT_MAX - 2U) // Reset stats after some time
+	{
+		StrobeInfo.pCounter = 0; StrobeInfo.pBCounter = 0; StrobeInfo.pNCounter = 0;
+		StrobeInfo.nCounter = 0; StrobeInfo.nBCounter = 0; StrobeInfo.nNCounter = 0;
+		StrobeInfo.fCounter = 0;
+		StrobeInfo.deviation = 0.0;
+		Q_memset(delta, 0, ARRAYSIZE(delta));
+		StrobeInfo.frameInfo &= ~p_inverted;
+	}
+	strobeInterval = r_strobe->integer;
+	swapInterval = r_strobe_swapinterval->integer;
+
+	if ( (strobeInterval == 0) ||
+		((gl_swapInterval->integer == 0) && (strobeInterval != 0)) )
```

Comment:
Integers are always FALSE when equal to zero and TRUE otherwise.

Reply:
I always try to be more explicit, but I noted it.

Reply:
It isn't bad. Just more readable for typical C or C++ programmer, I think. 
It will not readable for Java programmers. :D :D